subt/tools/run.py

```diff
@@ -0,0 +1,319 @@
+import argparse
+import datetime
+import os
+import pathlib
+import re
+import signal
+import subprocess
+import time
+
+import docker
+import docker.errors
+import docker.types
+import toml
+
+from pprint import pprint
+
+WORLDS=dict(
+    #TUNNEL
+    tq  = "tunnel_qual_ign",
+    ts1 = "simple_tunnel_01",
+    ts2 = "simple_tunnel_02",
+    ts3 = "simple_tunnel_03",
+    tp1 = "tunnel_circuit_practice_01",
+    tp2 = "tunnel_circuit_practice_02",
+    tp3 = "tunnel_circuit_practice_03",
+    tc1 = "tunnel_circuit_01",
+    tc7 = "tunnel_circuit_07",
+    tc12 = "tunnel_circuit_12",
+    tc13 = "tunnel_circuit_13",
+    tc15 = "tunnel_circuit_15",
+
+    # URBAN
+    uq  = "urban_qual",
+    us1 = "simple_urban_01",
+    us2 = "simple_urban_02",
+    us3 = "simple_urban_03",
+    up1 = "urban_circuit_practice_01",
+    up2 = "urban_circuit_practice_02",
+    up3 = "urban_circuit_practice_03",
+    uc1 = "urban_circuit_01",
+    uc2 = "urban_circuit_02",
+    uc3 = "urban_circuit_03",
+    uc4 = "urban_circuit_04",
+    uc5 = "urban_circuit_05",
+    uc6 = "urban_circuit_06",
+    uc7 = "urban_circuit_07",
+    uc8 = "urban_circuit_08",
+
+    #CAVE
+    cq  = "cave_qual",
+    cs1 = "simple_cave_01",
+    cs2 = "simple_cave_02",
+    cs3 = "simple_cave_03",
+    cp1 = "cave_circuit_practice_01",
+    cp2 = "cave_circuit_practice_02",
+    cp3 = "cave_circuit_practice_03",
+)
+
+ROBOTS=dict(
+    teambase = "TEAMBASE",
+    drone = "SSCI_X4_SENSOR_CONFIG_2",
+    freyja = "ROBOTIKA_FREYJA_SENSOR_CONFIG_2",
+)
+
+XAUTH = pathlib.Path("/tmp/.docker.xauth")
+
+
+def validate_world(world):
+    try:
+        return WORLDS[world]
+    except KeyError:
+        sys.exit(f"'{world}' not valid world identification")
+
+
+def validate_robots(robots, timeout):
+    valid = {}
+    for name, kind in robots.items():
+        try:
+            valid[name] = ROBOTS[kind]
+        except KeyError:
+            sys.exit(f"'{kind}' not valid robot kind identification")
+    #valid[f"T{timeout}"] = "TEAMBASE"
```

Comment:
this should or should not be here? - probably copy from `submit`?

Reply:
it is not implemented yet - multiple robots are TODO